test/mpi/dtpools/src/dtpools_custom.c

```diff
@@ -220,8 +220,13 @@ static int custom_hvector(DTP_pool_s dtp, DTPI_Attr_s * attr, MPI_Datatype * new
     n = sscanf(s, "numblks %d, blklen %d, stride %zd", &attr->u.hvector.numblks,
                &attr->u.hvector.blklen, &attr->u.hvector.stride);
     DTPI_ERR_ASSERT(n == 3, rc);
+#if MTEST_HAVE_MIN_MPI_VERSION(2,0)
+    rc = MPI_Type_create_hvector(attr->u.hvector.numblks, attr->u.hvector.blklen, attr->u.hvector.stride,
+                                 type, newtype);
+#else
     rc = MPI_Type_hvector(attr->u.hvector.numblks, attr->u.hvector.blklen, attr->u.hvector.stride,
                           type, newtype);
+#endif
```

Comment:
I would suggest just use the new function, i.e. `MPI_Type_create_hvector`. `MPI-1` implementations can always use older testsuite releases.